sdk/core/azure-core/src/test/java/com/azure/core/credential/TokenCacheTests.java

```diff
@@ -169,6 +170,27 @@ public void testLongRunningWontOverflow() throws Exception {
         assertTrue(refreshes.get() <= 11);
     }
 
+    @Test
+    public void testRefreshOnFlow() throws InterruptedException {
+        AtomicLong refreshes = new AtomicLong(0);
+
+        TokenCredential dummyCred = request -> {
+            refreshes.incrementAndGet();
+            return Mono.just(new TokenCacheTests.Token("testToken", 200000, 33000));
+        };
+
+        AccessTokenCache cache = new AccessTokenCache(dummyCred);
+
+        cache.getTokenSync(new TokenRequestContext(), false);
+
+        sleep(33000);
```

Comment:
Can we use something like https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html instead of having a test that waits for 33 seconds?